internal/command/jsonformat/computed/renderers/json.go

```diff
@@ -0,0 +1,57 @@
+package renderers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/hashicorp/terraform/internal/command/format"
+	"github.com/hashicorp/terraform/internal/command/jsonformat/computed"
+	"github.com/hashicorp/terraform/internal/command/jsonformat/jsondiff"
+	"github.com/hashicorp/terraform/internal/plans"
+)
+
+func DefaultJsonOpts() jsondiff.JsonOpts {
+	return jsondiff.JsonOpts{
+		Primitive: func(before, after interface{}, ctype cty.Type, action plans.Action) computed.Diff {
+			return computed.NewDiff(Primitive(before, after, ctype), action, false)
+		},
+		Object: func(elements map[string]computed.Diff, action plans.Action) computed.Diff {
+			return computed.NewDiff(Object(elements), action, false)
+		},
+		Array: func(elements []computed.Diff, action plans.Action) computed.Diff {
+			return computed.NewDiff(List(elements), action, false)
+		},
+		TypeChange: func(before, after computed.Diff, action plans.Action) computed.Diff {
+			return computed.NewDiff(TypeChange(before, after), action, false)
+		},
+	}
+}
+
+func (renderer primitiveRenderer) renderStringDiffAsJson(diff computed.Diff, indent int, opts computed.RenderHumanOpts, before evaluatedString, after evaluatedString) string {
```

Comment:
I've added a comment explaining why DefaultJsonOpts is where it is. It's because of cycles, and it's also referenced from within this package. Basically we want to call the JSON functions from this package, but we want the JSON functions to create renderers, which is a cycle. This callback loop gets around that. I've also renamed it to be RendererJsonOpts as default does seem to suggest it should be in the JSON package.

I've also moved all the primitiveRenderer functions into primitive.go. I had moved these out as they are large functions and I wanted to keep the files small but I think you're right that it's clearer having them grouped by class instead of by naming.